woocommerce/woocommerce
# Fix/31582 special characters in password on checkout page lead to not working account

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces `wp_slash()` on `$_POST['password']` to mimic the behavior of default account creation on registration during checkout via submitting a password.
On the default account creation, passwords are being slashed before encryption. Lack of this step would prevent a new user from subsequently logging in after creating an account during checkout while submitting the correct password

Closes #31582 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/41a96ebf-9053-47e8-8b8d-2da1e3c8b033">


1. Set up WooCommerce using the settings from the above printscreen on `/wp-admin/admin.php?page=wc-settings&tab=account`
2. On an incognito browser window, place an order using the shortcode checkout page, and create a new account while doing it using a password containing the `"` character
3. Go to My Account, log out, and try to log back in using the new user credentials
4. Verify that the login process was done successfully



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
